common/rpc.go

```diff
@@ -65,3 +69,31 @@ type (
 		GetMaxMessageSize() int
 	}
 )
+
+// GetClientHeaders returns the headers that should be sent from client to server
+func GetClientHeaders(ctx context.Context) map[string]string {
+	call := yarpc.CallFromContext(ctx)
+	headerNames := call.HeaderNames()
+	headerExists := map[string]struct{}{}
+	for _, h := range headerNames {
+		headerExists[h] = struct{}{}
+	}
+
+	headers := make(map[string]string)
+	if _, ok := headerExists[LibraryVersionHeaderName]; !ok {
+		headers[LibraryVersionHeaderName] = call.Header(LibraryVersionHeaderName)
+	}
+	if _, ok := headerExists[FeatureVersionHeaderName]; !ok {
+		headers[FeatureVersionHeaderName] = call.Header(FeatureVersionHeaderName)
+	}
+	if _, ok := headerExists[ClientImplHeaderName]; !ok {
+		headers[ClientImplHeaderName] = call.Header(ClientImplHeaderName)
+	}
+	if _, ok := headerExists[ClientFeatureFlagsHeaderName]; !ok {
+		headers[ClientFeatureFlagsHeaderName] = call.Header(ClientFeatureFlagsHeaderName)
+	}
+	if _, ok := headerExists[ClientIsolationGroupHeaderName]; !ok {
+		headers[ClientIsolationGroupHeaderName] = call.Header(ClientIsolationGroupHeaderName)
+	}
+	return headers
```

Comment:
this is an important part that I have some knowledge gap. 
1. do we forward headers to workflows in sync calls?  
2. by doing this allowlist based header selection would the user be disappointed when they don't see some of the custom headers in their workflow's headers?

Reply:
These are the headers either forwarded in sync calls or consumed in sync calls.
For example, isolation group header is used in sync calls for zonal isolation.

Reply:
@Groxx  I believe you were looking at reducing the set of headers that were either persisted or forwarded, waht's the current status of that?